src/Datadog.Trace/Agent/Api.cs

```diff
@@ -93,12 +97,32 @@ public async Task SendTracesAsync(IList<List<Span>> traces)
                     using (var content = new MsgPackContent<IList<List<Span>>>(traces, SerializationContext))
                     {
                         content.Headers.Add(AgentHttpHeaderNames.TraceCount, traceIds.Count.ToString());
-                        responseMessage = await _client.PostAsync(_tracesEndpoint, content).ConfigureAwait(false);
+
+                        try
+                        {
+                            _dogStatsdClient?.Increment(TracerMetricNames.Api.Requests);
+                            responseMessage = await _client.PostAsync(_tracesEndpoint, content).ConfigureAwait(false);
+                        }
+                        catch
+                        {
+                            // count the exceptions throw by the HttpClient,
+                            // not responses with 5xx status codes
+                            // (which cause EnsureSuccessStatusCode() to throw below)
+                            _dogStatsdClient?.Increment(TracerMetricNames.Api.Errors);
+                            throw;
+                        }
+
+                        // count every response's status code
+                        string[] tags = { $"status:{(int)responseMessage.StatusCode}" };
+                        _dogStatsdClient?.Increment(TracerMetricNames.Api.ResponsesByStatusCode, tags: tags);
+
                         responseMessage.EnsureSuccessStatusCode();
                     }
                 }
                 catch (Exception ex)
                 {
+                    _dogStatsdClient?.Increment(TracerMetricNames.Api.Errors);
```

Comment:
Oh, I get it -- by catching the earlier thrown exceptions you can count the overall `Api` errors, is that the idea?

Reply:
Oops, no, this was a mistake. Thanks for point it out!

If we are able to make the API request to the Agent and get a response, we don't count that as an error for purposes of the metrics. The only errors we want to count in this metrics are the ones thrown by `HttpClient.PostAsync()`, not the ones from `HttpResponseMessage.EnsureSuccessStatusCode()`.

All responses get counted below and grouped by status code (which accounts for 5xx error codes).

Fixed in 0510f08a1cf85d4dd3156939358df0a0b8c177e0.